src/engraving/libmscore/textbase.cpp

```diff
@@ -2980,7 +2980,7 @@ void TextBase::initTextStyleType(TextStyleType tid, bool preserveDifferent)
     } else {
         setTextStyleType(tid);
         for (const auto& p : *textStyle(tid)) {
-            if (getProperty(p.pid) == propertyDefault(p.pid)) {
```

Comment:
It looks like this code has been here for years (https://github.com/musescore/MuseScore/pull/4094). What exactly is the problem? Also, the previous implementation looks more logical: use the style value only when the property has a default value

Reply:
> It looks like this code has been here for years (#4094). What exactly is the problem? Also, the previous implementation looks more logical: use the style value only when the property has a default value

But why? All the default pros defined in style setting as far as I understood. And if you want initialise element with value other than default this ### if ### has no sense at all. Or am I wrong?

The problem is exactly about that. When we initialising Marker during GP import it initialised with wrong font size

Reply:
We're only string properties if they deviate from the default, so yes, that old code seems just wrong.
I do wonder though why this has never caused issues so far

Reply:
@Jojo-Schmitz 
Probably because we setting up element directly when creating it.
So looks like initTextStyleType actually makes no changes to element.

https://github.com/musescore/MuseScore/blob/04972b306c7a111252eb6331d716c1e3dcb99394/src/engraving/libmscore/marker.cpp#L54

```cpp
Marker::Marker(EngravingItem* parent, TextStyleType tid)
    : TextBase(ElementType::MARKER, parent, tid, ElementFlag::MOVABLE | ElementFlag::ON_STAFF | ElementFlag::SYSTEM)
{
    initElementStyle(&markerStyle);
    _markerType = MarkerType::FINE;
    setLayoutToParentWidth(true);
}
```

```cpp
void TextBase::initElementStyle(const ElementStyle* ss)
{
    m_elementStyle = ss;
    size_t n      = ss->size() + TEXT_STYLE_SIZE;

    delete[] m_propertyFlagsList;
    m_propertyFlagsList = new PropertyFlags[n];
    for (size_t i = 0; i < n; ++i) {
        m_propertyFlagsList[i] = PropertyFlags::STYLED;
    }
    for (const StyledProperty& p : *m_elementStyle) {
        setProperty(p.pid, styleValue(p.pid, p.sid));
    }
    for (const auto& p : *textStyle(textStyleType())) {
        setProperty(p.pid, styleValue(p.pid, p.sid));
    }
}
```